framework/platform/bpl/uci/db/bpl_db_uci.cpp

```diff
@@ -44,7 +44,48 @@ bool uci_section_exists(const std::string &package_name, const std::string &sect
     LOG(TRACE) << "uci_section_exists() " << package_name << ".(" << section_type << ")"
                << section_name;
 
-    return true;
+    auto ctx = alloc_context();
+    if (!ctx) {
+        return false;
+    }
+
+    char path[MAX_UCI_BUF_LEN] = {0};
+    if (snprintf(path, MAX_UCI_BUF_LEN, package_path, package_name.c_str()) <= 0) {
+        LOG(ERROR) << "Failed to compose path";
+        return false;
+    }
+
+    uci_package *pkg = nullptr;
+    if (!(pkg = uci_lookup_package(ctx.get(), path))) {
+        // Try and load the package file, in case it wasn't loaded before
+        uci_load(ctx.get(), path, &pkg);
+        if (!pkg) {
+            LOG(ERROR) << "uci lookup package failed!" << std::endl << uci_get_error(ctx.get());
+            return false;
+        }
+    }
+
+    bool found = false;
+    // Loop through the sections with a matching section name (section_name)
+    uci_element *elm = nullptr;
+    uci_foreach_element(&pkg->sections, elm)
+    {
+        found            = true;
+        uci_section *sec = uci_to_section(elm);
+        // if the section_type is empty all matching sections are a valid match
+        auto name_match = section_name.compare(sec->e.name) == 0;
+        auto type_match = section_type.empty() || section_type.compare(sec->type) == 0;
+        if (name_match && type_match) {
+            LOG(TRACE) << "Found match: " << sec->type << "(" << sec->e.name << ")";
+            return true;
+        }
+    }
+
+    LOG(TRACE) << "No " << section_name << " section found"
+               << (section_type.empty() || !found ? "!"
+                                                  : std::string(" with type: ") + section_type);
+
+    return false;
```

Comment:
```suggestion
    bool section_found  = false;
    // Loop through the sections with a matching section name (section_name)
    uci_element *elm = nullptr;
    uci_section *sec = nullptr;
    uci_foreach_element(&pkg->sections, elm)
    {
        uci_section *sec = uci_to_section(elm);
        // uci enforces sections must have unique name
        if (section_name.compare(sec->e.name) == 0) {
            section_found = true;
            break;
        }
    }
    
    if (!section_found) {
        LOG(DEBUG) << "Section not found for " << section_name;
        return false;
    }
    
    if (!section_type.empty() && !(section_type.compare(sec->type) == 0) {
        LOG(WARNING) << "Section type missmatch for " << section_name << ", requested-type=" << section_type << ", actual-type=" << sec->type;
        return false;
    }

    return true;
```

no special print is required for the successful case - since this might be used in many places and successive cases provide no additional info.

Reply:
Refactored the function:
- Removed the successful case print
- Untangled the nested conditions